mitodl/micromasters
# Add API for getting information about AutomaticEmails

#### What are the relevant tickets?

closes #2970 

#### What's this PR do?

this adds an API for AutomaticEmails and handled the GET request for it.

#### How should this be manually tested?

Easy way:

1. add some AutomaticEmails (by sending learner search emails and checking the little box)
2. edit `DashboardPage.js` and add this at the top of the file:

```js
import * as api from '../lib/api';

window.api = api;
```

3. go to `/dashboard`, type this in the JS console, and confirm you get some intelligible response:

```js
api.fetchJSONWithCSRF("/api/v0/mail/automatic_email/").then(console.log);
```
